### DISCUSSION TITLE
Can I use a "user directory" variable in Freeplane links??

### DISCUSSION THREAD

**Discussion by EdoFro** [freeplane/freeplane#573] [Help] [answered]
Something like a node with such a link:

`%userfpdir%/resources/images/jumper.png`

and hopefully it can be used in node's links, but also in attribute's links. Even better if it can be also used in links inside html or markdown in node's texts, details or notes.

anything like that?

thanks in advance,

edo

**Comment by macmarrum** [comment, marked as answer]
I know what you mean. Something like [#at()](https://www.freeplane.org/doc/api/org/freeplane/api/NodeRO.html#at-java.lang.String-) in links, but for Freeplane variables. Maybe even wrapping `System.getenv(variableName)` too.

I believe that some of your requirements can be covered by using scripts/formulas, e.g.

```groovy
="${org.freeplane.core.resources.ResourceController.resourceController.freeplaneUserDirectory}/resources/images/jumper.png"
```

```groovy
node.link.file = new File("${org.freeplane.core.resources.ResourceController.resourceController.freeplaneUserDirectory}/resources/images/jumper.png")
```

```groovy
="""\
# Markdown Test
![](${org.freeplane.core.resources.ResourceController.resourceController.freeplaneUserDirectory}/resources/images/jumper.png})
"""
```

**Reply by EdoFro** [reply]
Great. Thanks, but I was searching a way to access it without having to execute a script or formula.

**Reply by dpolivaev** [reply]
I think that the answer given in #568 works also here.

**Reply by EdoFro** [reply]
specifically, this comment:
https://github.com/freeplane/freeplane/discussions/568#discussioncomment-3432960
